### PR TITLE
Synonyms API - Get Synonym Rule

### PR DESCRIPTION
**Pull request by carlosdelest**
Allows to retrieve an individual synonym rule from a synonym set:

`GET /_synonyms/<synonym_set_id>/<synonym_rule_id>`

Response:

```json
{
    "id": "test-2",
    "synonyms": "test => check"
}
```

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/synonyms/TransportGetSynonymsAction.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ public TransportGetSynonymsAction(TransportService transportService, ActionFilte
 
     @Override
     protected void doExecute(Task task, GetSynonymsAction.Request request, ActionListener<GetSynonymsAction.Response> listener) {
-        synonymsManagementAPIService.getSynonymRules(
+        synonymsManagementAPIService.getSynonymSetRules(
```
Comment (carlosdelest) [inline server/src/main/java/org/elasticsearch/action/synonyms/TransportGetSynonymsAction.java:33]: I figured changing this name made the intention clearer, as we now have synonym set rules and single synonym rule

File: server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java [hunk 2]
```diff
@@ -170,30 +169,29 @@ public void onFailure(Exception e) {
             });
     }
 
-    public void getSynonymRules(String resourceName, int from, int size, ActionListener<PagedResult<SynonymRule>> listener) {
+    public void getSynonymSetRules(String synonymSetId, int from, int size, ActionListener<PagedResult<SynonymRule>> listener) {
```
Comment (carlosdelest) [inline server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java:172]: Minor param refactoring

File: server/src/main/java/org/elasticsearch/action/synonyms/GetSynonymRuleAction.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            return Objects.hash(synonymsSetId, synonymRuleId);
+        }
+    }
+
+    public static class Response extends ActionResponse implements ToXContentObject {
+
+        private final SynonymRule synonymRule;
+
+        public Response(StreamInput in) throws IOException {
+            this.synonymRule = new SynonymRule(in);
```
Comment (kderusso) [inline server/src/main/java/org/elasticsearch/action/synonyms/GetSynonymRuleAction.java:98]: Do we need to call `super(in)` here before setting the synonym rule?

File: server/src/main/java/org/elasticsearch/action/synonyms/GetSynonymRuleAction.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        }
+
+        public Response(SynonymRule synonymRule) {
+            this.synonymRule = synonymRule;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            synonymRule.writeTo(out);
```
Comment (kderusso) [inline server/src/main/java/org/elasticsearch/action/synonyms/GetSynonymRuleAction.java:107]: Do we need to call `super.writeTo(out)` here?

**Review by kderusso — APPROVED** [review]
Looks good to me!

File: server/src/main/java/org/elasticsearch/action/synonyms/GetSynonymRuleAction.java (hunk 3 above)
Comment (carlosdelest) [reply server/src/main/java/org/elasticsearch/action/synonyms/GetSynonymRuleAction.java:99]: Good catch, thanks!

File: server/src/main/java/org/elasticsearch/action/synonyms/GetSynonymRuleAction.java (hunk 4 above)
Comment (carlosdelest) [reply server/src/main/java/org/elasticsearch/action/synonyms/GetSynonymRuleAction.java:108]: Thanks for checking! We can't, it's an abstract method for the super class.

**Review by kderusso — APPROVED** [review]

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/60_synonym_rule_get.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
+setup:
+  - skip:
+      version: " - 8.8.99"
```
Comment (mayya-sharipova) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/60_synonym_rule_get.yml:3]: We need to put 8.9.99 for "Introduced in 8.10.0" ( at least TODO if this version is not yet available)

**Review by mayya-sharipova — APPROVED** [review]
@carlosdelest Thanks Carlos, as always great work!

File: server/src/main/java/org/elasticsearch/action/synonyms/GetSynonymRuleAction.java (hunk 4 above)
Comment (carlosdelest) [reply server/src/main/java/org/elasticsearch/action/synonyms/GetSynonymRuleAction.java:108]: Yes, that's for `Request`, not `Response`. They have different base classes 👍 . The call for super.writeTo(out) is present on the Request 👍

**Pull request merged**